tests/jax/components/building/adders_test.py

```diff
@@ -15,135 +15,211 @@
-        )
-        return components, {}
+    signature = ParallelSequenceAdderSignature()
+    return signature
 
 
 @pytest.fixture
-def test_system_parallel_sequence_adder() -> System:
-    """Dummy system with zero components."""
-    return TestSystemWithParallelSequenceAdder()
+def parallel_transition_adder_signature() -> ParallelTransitionAdderSignature:
+    """Creates a parallel transition adder signature fixture
+
+    Returns:
+        ParallelTransitionAdderSignature.
+    """
+
+    signature = ParallelTransitionAdderSignature()
+    return signature
 
 
-# TODO Fix test.
-def test_adders(
-    test_system_parallel_sequence_adder: System,
+@pytest.fixture
+def uniform_priority() -> UniformAdderPriority:
+    """Creates a uniform adder priority fixture
+
+    Returns:
+        UniformAdderPriority.
+    """
+
+    priority = UniformAdderPriority()
+    return priority
+
+
+def test_sequence_adders(
+    mock_builder: Builder,
+    parallel_sequence_adder: ParallelSequenceAdder,
 ) -> None:
-    """Test if system builder instantiates processes as expected."""
-    # TODO Update once rewrite these tests.
-    test_system_parallel_sequence_adder.build(
-        environment_factory=make_fake_environment_factory()
+    """Test sequence adder callbacks.
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_sequence_adder: Fixture ParallelSequenceAdder.
+
+    Returns:
+        None
+    """
+
+    parallel_sequence_adder.on_building_init_start(builder=mock_builder)
+    assert (
+        mock_builder.store.sequence_length
+        == parallel_sequence_adder.config.sequence_length
     )
-    test_system_parallel_sequence_adder._builder.data_server()
-    test_system_parallel_sequence_adder._builder.store.system_executor = None
-    test_system_parallel_sequence_adder._builder.executor(
-        executor_id="executor", data_server_client=None, parameter_server_client=None
+    parallel_sequence_adder.on_building_executor_adder(builder=mock_builder)
+    assert type(mock_builder.store.adder) == reverb_adders.ParallelSequenceAdder
+    assert parallel_sequence_adder.name() == "executor_adder"
+
+
+def test_sequence_adders_signature(
+    mock_builder: Builder,
+    parallel_sequence_adder_signature: ParallelSequenceAdderSignature,
+) -> None:
+    """Test sequence adder signature callback.
+
+    Args:
+        mock_builder: Fixture SystemBuilder.
+        parallel_sequence_adder_signature: Fixture ParallelSequenceAdderSignature.
+
+    Returns:
+        None
+    """
+    parallel_sequence_adder_signature.on_building_data_server_adder_signature(
+        builder=mock_builder
+    )
+    assert mock_builder.store.adder_signature_fn
+    assert parallel_sequence_adder_signature.name() == "data_server_adder_signature"
+    assert issubclass(
+        parallel_sequence_adder_signature.config_class(),  # type: ignore
+        AdderSignatureConfig,
+    )
+
+
+def test_transition_adders(
```

Comment:
`test_parallel_transition_adder`. Same principle applies elsewhere: i.e. one test per component, not per group of components.